File: examples/echo/src/main/java/com/twitter/jexample/EchoLoadTestConfig.java
package com.twitter.jexample;

import com.twitter.iago.ParrotServerFlags;
import com.twitter.iago.ThriftParrotConfig;
import com.twitter.iago.processor.RecordProcessor;

public class EchoLoadTestConfig extends ThriftParrotConfig {
  public EchoLoadTestConfig(ParrotServerFlags config) {
    super(config, true);
  }

  @Override
  public RecordProcessor recordProcessor() {
    return new EchoLoadTest(service());
  }
}

File: examples/echo/src/main/scala/com/twitter/example/EchoLoadTestConfig.scala
package com.twitter.example

import com.twitter.iago.{ParrotServerFlags, ThriftParrotConfig}
import com.twitter.iago.util.{RequestDistribution, SlowStartPoissonProcess}
import com.twitter.util.TimeConversions._

class EchoLoadTestConfig(config: ParrotServerFlags) extends ThriftParrotConfig(config) {
  override val recordProcessor = new EchoLoadTest(service)
  override def distribution(rate: Int): RequestDistribution = new SlowStartPoissonProcess(rate, 1.minute)
}
